util/src/main/DocTaglet.java

```diff
@@ -14,69 +14,66 @@
  * limitations under the License.
  */
 
-import com.sun.javadoc.Tag;
-import com.sun.tools.doclets.Taglet;
 
-public abstract class DocTaglet implements Taglet {
-
-    public boolean inConstructor() {
-        return true;
-    }
-
-    public boolean inField() {
-        return true;
-    }
-
-    public boolean inMethod() {
-        return true;
-    }
+import com.sun.source.doctree.DocTree;
+import jdk.javadoc.doclet.Taglet;
 
-    public boolean inOverview() {
-        return true;
-    }
+import javax.lang.model.element.Element;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
-    public boolean inPackage() {
-        return true;
-    }
+public abstract class DocTaglet implements Taglet {
 
-    public boolean inType() {
-        return true;
+    @Override
+    public Set<Location> getAllowedLocations() {
+        return new HashSet<Location>(Arrays.asList(
+                Location.CONSTRUCTOR,
+                Location.METHOD,
+                Location.FIELD,
+                Location.OVERVIEW,
+                Location.PACKAGE,
+                Location.TYPE));
     }
 
     public boolean isInlineTag() {
         return false;
     }
 
-    public String toString(final Tag[] tags) {
-        if (tags.length == 0) {
-            return null;
-        }
-
-        StringBuilder buf = new StringBuilder(String.format("\n<dl><dt><span class=\"strong\">%s</span></dt>\n", getHeader()));
-        for (Tag t : tags) {
-            buf.append("   <dd>").append(genLink(t.text())).append("</dd>\n");
+    @Override
+    public String toString(final List<? extends DocTree> docTreeList, final Element ignored) {
+        StringBuilder buf = new StringBuilder();
+        buf.append("<dl>\n");
+        buf.append(String.format("   <dt><span class=\"strong\">%s</span></dt>\n", getHeader()));
+        for (DocTree t : docTreeList) {
+            buf.append("   <dd>").append(genLink(t.toString())).append("</dd>\n");
         }
+        buf.append("</dl>\n");
         return buf.toString();
     }
 
     protected abstract String getHeader();
 
-    public String toString(final Tag tag) {
-        return toString(new Tag[]{tag});
-    }
+    protected abstract String getBaseDocURI();
 
-    protected String genLink(final String text) {
-        String relativePath = text;
-        String display = text;
+    // Generate link from strings like:
```

Comment:
This got slightly more complicated because the new Taglet framework includes the custom tag name as part of the text, which means there's one more space character to consider.